### ISSUE TITLE
SpongeForge server crash while generating land

### ISSUE THREAD

**Issue by MJRamon** [Ivorforce/RecurrentComplex#364]
Minecraft 1.12.2
spongeforge-1.12.2-2768-7.1.4
RecurrentComplex-1.4.7

```
java.lang.Error: ServerHangWatchdog detected that a single server tick took 60.00 seconds (should be max 0.05)
	at net.minecraft.util.math.Vec3i.equals(SourceFile:39)
	at gnu.trove.impl.hash.TObjectHash.equals(TObjectHash.java:478)
	at gnu.trove.impl.hash.TObjectHash.index(TObjectHash.java:177)
	at gnu.trove.impl.hash.TObjectHash.contains(TObjectHash.java:153)
	at gnu.trove.map.hash.TObjectDoubleHashMap.containsKey(TObjectDoubleHashMap.java:210)
	at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerNaturalAir.lambda$transform$322(TransformerNaturalAir.java:113)
	at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerNaturalAir$$Lambda$2011/1614325801.test(Unknown Source)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:174)
	at java.util.Iterator.forEachRemaining(Iterator.java:116)
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:580)
	at java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:270)
	at java.util.Iterator.forEachRemaining(Iterator.java:116)
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
	at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerNaturalAir.transform(TransformerNaturalAir.java:115)
	at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerNaturalAir.transform(TransformerNaturalAir.java:49)
	at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerMulti.lambda$transform$24(TransformerMulti.java:185)
	at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerMulti$$Lambda$1997/1875487759.accept(Unknown Source)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerMulti.transform(TransformerMulti.java:185)
	at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerMulti.transform(TransformerMulti.java:41)
	at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerMulti.lambda$transform$24(TransformerMulti.java:185)
	at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerMulti$$Lambda$1997/1875487759.accept(Unknown Source)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerMulti.transform(TransformerMulti.java:185)
	at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerMulti.transform(TransformerMulti.java:41)
	at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerMulti.lambda$transform$24(TransformerMulti.java:185)
	at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerMulti$$Lambda$1997/1875487759.accept(Unknown Source)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerMulti.transform(TransformerMulti.java:185)
	at ivorius.reccomplex.world.gen.feature.structure.generic.GenericStructure.generate(GenericStructure.java:260)
	at ivorius.reccomplex.world.gen.feature.structure.generic.GenericStructure.generate(GenericStructure.java:74)
	at ivorius.reccomplex.world.gen.feature.StructureGenerator.generate(StructureGenerator.java:216)
	at ivorius.reccomplex.world.gen.feature.WorldGenStructures.planStructureInChunk(WorldGenStructures.java:120)
	at ivorius.reccomplex.world.gen.feature.WorldGenStructures.lambda$planStructuresInChunk$205(WorldGenStructures.java:77)
	at ivorius.reccomplex.world.gen.feature.WorldGenStructures$$Lambda$1918/718380578.accept(Unknown Source)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1382)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	at ivorius.reccomplex.world.gen.feature.WorldGenStructures.planStructuresInChunk(WorldGenStructures.java:77)
	at ivorius.reccomplex.world.gen.feature.WorldGenStructures.decorate(WorldGenStructures.java:222)
	at ivorius.reccomplex.events.handlers.RCForgeEventHandler.onPreChunkDecoration(RCForgeEventHandler.java:85)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_786_RCForgeEventHandler_onPreChunkDecoration_Pre.invoke(.dynamic)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90)
	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:182)
	at biomesoplenty.common.world.ChunkGeneratorOverworldBOP.func_185931_b(ChunkGeneratorOverworldBOP.java:498)
	at net.minecraft.world.chunk.Chunk.func_186034_a(Chunk.java:1019)
	at net.minecraft.world.chunk.Chunk.func_186030_a(Chunk.java:990)
	at net.minecraft.world.gen.ChunkProviderServer.func_186025_d(ChunkProviderServer.java:157)
	at net.minecraft.server.management.PlayerChunkMapEntry.func_187268_a(PlayerChunkMapEntry.java:126)
	at net.minecraft.server.management.PlayerChunkMap.func_72693_b(SourceFile:147)
	at net.minecraft.world.WorldServer.func_72835_b(WorldServer.java:227)
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:756)
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:397)
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:668)
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:526)
	at java.lang.Thread.run(Thread.java:748)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Server Watchdog
Stacktrace:
	at net.minecraft.util.math.Vec3i.equals(SourceFile:39)
	at gnu.trove.impl.hash.TObjectHash.equals(TObjectHash.java:478)
	at gnu.trove.impl.hash.TObjectHash.index(TObjectHash.java:177)
	at gnu.trove.impl.hash.TObjectHash.contains(TObjectHash.java:153)
	at gnu.trove.map.hash.TObjectDoubleHashMap.containsKey(TObjectDoubleHashMap.java:210)
	at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerNaturalAir.lambda$transform$322(TransformerNaturalAir.java:113)
	at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerNaturalAir$$Lambda$2011/1614325801.test(Unknown Source)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:174)
	at java.util.Iterator.forEachRemaining(Iterator.java:116)
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:580)
	at java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:270)
	at java.util.Iterator.forEachRemaining(Iterator.java:116)
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
	at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerNaturalAir.transform(TransformerNaturalAir.java:115)
	at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerNaturalAir.transform(TransformerNaturalAir.java:49)
	at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerMulti.lambda$transform$24(TransformerMulti.java:185)
	at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerMulti$$Lambda$1997/1875487759.accept(Unknown Source)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerMulti.transform(TransformerMulti.java:185)
	at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerMulti.transform(TransformerMulti.java:41)
	at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerMulti.lambda$transform$24(TransformerMulti.java:185)
	at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerMulti$$Lambda$1997/1875487759.accept(Unknown Source)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerMulti.transform(TransformerMulti.java:185)
	at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerMulti.transform(TransformerMulti.java:41)
	at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerMulti.lambda$transform$24(TransformerMulti.java:185)
	at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerMulti$$Lambda$1997/1875487759.accept(Unknown Source)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerMulti.transform(TransformerMulti.java:185)
	at ivorius.reccomplex.world.gen.feature.structure.generic.GenericStructure.generate(GenericStructure.java:260)
	at ivorius.reccomplex.world.gen.feature.structure.generic.GenericStructure.generate(GenericStructure.java:74)
	at ivorius.reccomplex.world.gen.feature.StructureGenerator.generate(StructureGenerator.java:216)
	at ivorius.reccomplex.world.gen.feature.WorldGenStructures.planStructureInChunk(WorldGenStructures.java:120)
	at ivorius.reccomplex.world.gen.feature.WorldGenStructures.lambda$planStructuresInChunk$205(WorldGenStructures.java:77)
	at ivorius.reccomplex.world.gen.feature.WorldGenStructures$$Lambda$1918/718380578.accept(Unknown Source)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1382)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	at ivorius.reccomplex.world.gen.feature.WorldGenStructures.planStructuresInChunk(WorldGenStructures.java:77)
	at ivorius.reccomplex.world.gen.feature.WorldGenStructures.decorate(WorldGenStructures.java:222)
	at ivorius.reccomplex.events.handlers.RCForgeEventHandler.onPreChunkDecoration(RCForgeEventHandler.java:85)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_786_RCForgeEventHandler_onPreChunkDecoration_Pre.invoke(.dynamic)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90)
	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:182)
	at biomesoplenty.common.world.ChunkGeneratorOverworldBOP.func_185931_b(ChunkGeneratorOverworldBOP.java:498)
	at net.minecraft.world.chunk.Chunk.func_186034_a(Chunk.java:1019)
	at net.minecraft.world.chunk.Chunk.func_186030_a(Chunk.java:990)
	at net.minecraft.world.gen.ChunkProviderServer.func_186025_d(ChunkProviderServer.java:157)
	at net.minecraft.server.management.PlayerChunkMapEntry.func_187268_a(PlayerChunkMapEntry.java:126)
	at net.minecraft.server.management.PlayerChunkMap.func_72693_b(SourceFile:147)
	at net.minecraft.world.WorldServer.func_72835_b(WorldServer.java:227)
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:756)
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:397)
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:668)

-- Thread Dump --
Details:
	Threads: "Netty Epoll Server IO #3" Id=82 RUNNABLE
	at io.netty.channel.epoll.Native.epollWait0(Native Method)
	at io.netty.channel.epoll.Native.epollWait(Native.java:117)
	at io.netty.channel.epoll.EpollEventLoop.epollWait(EpollEventLoop.java:223)
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:247)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)
	at java.lang.Thread.run(Thread.java:748)


"Chunk I/O Executor Thread-1" Id=80 WAITING on java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject@457497a5
	at sun.misc.Unsafe.park(Native Method)
	-  waiting on java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject@457497a5
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2039)
	at java.util.concurrent.LinkedBlockingQueue.take(LinkedBlockingQueue.java:442)
	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1074)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)


"Netty Epoll Server IO #2" Id=76 RUNNABLE
	at io.netty.channel.epoll.Native.epollWait0(Native Method)
	at io.netty.channel.epoll.Native.epollWait(Native.java:117)
	at io.netty.channel.epoll.EpollEventLoop.epollWait(EpollEventLoop.java:223)
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:247)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)
	at java.lang.Thread.run(Thread.java:748)


"threadDeathWatcher-2-1" Id=75 TIMED_WAITING
	at java.lang.Thread.sleep(Native Method)
	at io.netty.util.ThreadDeathWatcher$Watcher.run(ThreadDeathWatcher.java:152)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	at java.lang.Thread.run(Thread.java:748)


"Netty Epoll Server IO #1" Id=74 RUNNABLE (in native)
	at io.netty.channel.epoll.Native.epollWait0(Native Method)
	at io.netty.channel.epoll.Native.epollWait(Native.java:117)
	at io.netty.channel.epoll.EpollEventLoop.epollWait(EpollEventLoop.java:223)
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:247)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)
	at java.lang.Thread.run(Thread.java:748)


"BL Environment Event Overrides Downloader #0" Id=72 WAITING on java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject@4bb3717b
	at sun.misc.Unsafe.park(Native Method)
	-  waiting on java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject@4bb3717b
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2039)
	at java.util.concurrent.LinkedBlockingQueue.take(LinkedBlockingQueue.java:442)
	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1074)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)


"File IO Thread" Id=71 TIMED_WAITING
	at java.lang.Thread.sleep(Native Method)
	at net.minecraft.world.storage.ThreadedFileIOBase.func_75736_b(SourceFile:51)
	at net.minecraft.world.storage.ThreadedFileIOBase.run(SourceFile:30)
	at java.lang.Thread.run(Thread.java:748)


"Timer-0" Id=68 TIMED_WAITING on java.util.TaskQueue@4f8cfcd5
	at java.lang.Object.wait(Native Method)
	-  waiting on java.util.TaskQueue@4f8cfcd5
	at java.util.TimerThread.mainLoop(Timer.java:552)
	at java.util.TimerThread.run(Timer.java:505)


"Server Watchdog" Id=66 RUNNABLE
	at sun.management.ThreadImpl.dumpThreads0(Native Method)
	at sun.management.ThreadImpl.dumpAllThreads(ThreadImpl.java:454)
	at net.minecraft.server.dedicated.ServerHangWatchdog.run(ServerHangWatchdog.java:45)
	at java.lang.Thread.run(Thread.java:748)


"pool-3-thread-3" Id=65 WAITING on java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject@a8e7b40
	at sun.misc.Unsafe.park(Native Method)
	-  waiting on java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject@a8e7b40
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2039)
	at java.util.concurrent.LinkedBlockingQueue.take(LinkedBlockingQueue.java:442)
	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1074)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)


"pool-3-thread-2" Id=64 WAITING on java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject@a8e7b40
	at sun.misc.Unsafe.park(Native Method)
	-  waiting on java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject@a8e7b40
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2039)
	at java.util.concurrent.LinkedBlockingQueue.take(LinkedBlockingQueue.java:442)
	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1074)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)


"pool-3-thread-1" Id=63 WAITING on java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject@a8e7b40
	at sun.misc.Unsafe.park(Native Method)
	-  waiting on java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject@a8e7b40
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2039)
	at java.util.concurrent.LinkedBlockingQueue.take(LinkedBlockingQueue.java:442)
	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1074)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)


"Netty Epoll Server IO #0" Id=39 RUNNABLE
	at io.netty.channel.epoll.Native.epollWait0(Native Method)
	at io.netty.channel.epoll.Native.epollWait(Native.java:117)
	at io.netty.channel.epoll.EpollEventLoop.epollWait(EpollEventLoop.java:223)
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:247)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)
	at java.lang.Thread.run(Thread.java:748)


"JLine terminal non blocking reader thread" Id=25 RUNNABLE (in native)
	at java.io.FileInputStream.readBytes(Native Method)
	at java.io.FileInputStream.read(FileInputStream.java:255)
	at org.jline.utils.InputStreamReader.read(InputStreamReader.java:270)
	-  locked java.io.FileInputStream@6b260c0b
	at org.jline.utils.InputStreamReader.read(InputStreamReader.java:200)
	-  locked java.io.FileInputStream@6b260c0b
	at org.jline.utils.NonBlockingReader.run(NonBlockingReader.java:276)
	at java.lang.Thread.run(Thread.java:748)


"Server console handler" Id=24 TIMED_WAITING on org.jline.utils.NonBlockingReader@3bce90bd
	at java.lang.Object.wait(Native Method)
	-  waiting on org.jline.utils.NonBlockingReader@3bce90bd
	at org.jline.utils.NonBlockingReader.read(NonBlockingReader.java:174)
	at org.jline.utils.NonBlockingReader.read(NonBlockingReader.java:113)
	at org.jline.keymap.BindingReader.readCharacter(BindingReader.java:132)
	at org.jline.keymap.BindingReader.readBinding(BindingReader.java:109)
	at org.jline.keymap.BindingReader.readBinding(BindingReader.java:60)
	at org.jline.reader.impl.LineReaderImpl.readBinding(LineReaderImpl.java:724)
	at org.jline.reader.impl.LineReaderImpl.readLine(LineReaderImpl.java:526)
	...


"DestroyJavaVM" Id=23 RUNNABLE


"Server thread" Id=21 RUNNABLE
	at net.minecraft.util.math.Vec3i.equals(SourceFile:39)
	at gnu.trove.impl.hash.TObjectHash.equals(TObjectHash.java:478)
	at gnu.trove.impl.hash.TObjectHash.index(TObjectHash.java:177)
	at gnu.trove.impl.hash.TObjectHash.contains(TObjectHash.java:153)
	at gnu.trove.map.hash.TObjectDoubleHashMap.containsKey(TObjectDoubleHashMap.java:210)
	at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerNaturalAir.lambda$transform$322(TransformerNaturalAir.java:113)
	at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerNaturalAir$$Lambda$2011/1614325801.test(Unknown Source)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:174)
	...


"Server Infinisleeper" Id=20 TIMED_WAITING
	at java.lang.Thread.sleep(Native Method)
	at net.minecraft.server.dedicated.DedicatedServer$1.run(DedicatedServer.java:82)


"Snooper Timer" Id=19 TIMED_WAITING on java.util.TaskQueue@6aa1840f
	at java.lang.Object.wait(Native Method)
	-  waiting on java.util.TaskQueue@6aa1840f
	at java.util.TimerThread.mainLoop(Timer.java:552)
	at java.util.TimerThread.run(Timer.java:505)


"Signal Dispatcher" Id=5 RUNNABLE


"Finalizer" Id=3 WAITING on java.lang.ref.ReferenceQueue$Lock@31bc8b04
	at java.lang.Object.wait(Native Method)
	-  waiting on java.lang.ref.ReferenceQueue$Lock@31bc8b04
	at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:144)
	at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:165)
	at java.lang.ref.Finalizer$FinalizerThread.run(Finalizer.java:216)


"Reference Handler" Id=2 WAITING on java.lang.ref.Reference$Lock@7e38aef6
	at java.lang.Object.wait(Native Method)
	-  waiting on java.lang.ref.Reference$Lock@7e38aef6
	at java.lang.Object.wait(Object.java:502)
	at java.lang.ref.Reference.tryHandlePending(Reference.java:191)
	at java.lang.ref.Reference$ReferenceHandler.run(Reference.java:153)



Stacktrace:
	at net.minecraft.server.dedicated.ServerHangWatchdog.run(ServerHangWatchdog.java:62)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by Ivorforce** [comment]
Duplicate of #155. I recommend pushing down the structureSpawnChance and / or tree decoration replacement chance down in the config for now. A fix will, hopefully, soon follow.
